cognite_toolkit/_cdf_tk/warnings.py

```diff
@@ -52,3 +64,40 @@ def __str__(self) -> str:
             for warning in group_list:
                 output.append(f"{'    ' * 2} * {warning!s}")
         return "\n".join(output)
+
+
+@dataclass(frozen=True)
+class GeneralWarning(ToolkitWarning):
+    severity: SeverityLevel | None = None
+    message: str | None = None
+    details: Union[None, str, list[str]] = None  # Allow None, str, list[str]
+
+    def __str__(self) -> str:
+        output = [f"{RICH_WARNING_FORMAT}{self.severity}{type(self).__name__}: {self.message}"]
+
+        if self.details:
+            if isinstance(self.details, str):
+                output.append(f"{'    ' * 2}{self.details}")
+            else:
+                for detail in self.details:
+                    output.append(f"{'    ' * 2}{detail}")
+        return "\n".join(output)
+
+
+@dataclass(frozen=True)
+class ToolkitDependenciesIncludedWarning(GeneralWarning):
+    severity: SeverityLevel = SeverityLevel.LOW
+    message: str = "Some resources were added due to dependencies."
+    details: Union[None, str, list[str]] = None  # Allow None, str, list[str]
+
+    def __init__(self, details: Union[None, str, list[str]] = None) -> None:
+        super().__init__(message=self.message, details=details, severity=self.severity)
+
+
+@dataclass(frozen=True)
+class ToolkitCleanDatasetNotSupportedWarning(GeneralWarning):
```

Comment:
Maybe a less specific warning?
```suggestion
class ToolkitNotSupportedWarning(GeneralWarning):
    functionality: str
```

Reply:
Then, instantiate it with `ToolkitNotSupportedWarning("Dataset cleaning")`?